inc/class-book.php

```diff
@@ -89,34 +89,49 @@ static function getBookInformation( $id = null, $contributors_as_string = true )
 			global $blog_id;
 		}
 
+		// ----------------------------------------------------------------------------
+		// Book Information
+		// ----------------------------------------------------------------------------
+
+		$book_information = [];
+		$meta = new Metadata();
+		$meta_post = $meta->getMetaPost();
+
+		$contributors = new Contributors();
+
 		// -----------------------------------------------------------------------------
 		// Is cached?
 		// -----------------------------------------------------------------------------
 
 		$cache_id = "book-inf-$blog_id";
 		if ( static::useCache() ) {
-			$book_information = wp_cache_get( $cache_id, 'pb' );
-			if ( $book_information ) {
-				return $book_information;
+			$cached_book_information = wp_cache_get( $cache_id, 'pb' );
+			if ( $cached_book_information ) {
+				if ( ! $read_contributors_from_cache ) {
+					$cached_book_information = array_merge(
+						$cached_book_information,
+						$contributors->getAll(
+							$meta_post->ID,
+							$contributors_as_string,
+							true
+						)
+					);
+				}
+				return $cached_book_information;
 			}
 		}
 
-		// ----------------------------------------------------------------------------
-		// Book Information
-		// ----------------------------------------------------------------------------
-
-		$book_information = [];
-		$meta = new Metadata();
-		$meta_post = $meta->getMetaPost();
-
 		if ( $meta_post ) {
 
 			// Contributors
-			$contributors = new Contributors();
-			$all_contributors = $contributors->getAll( $meta_post->ID, $contributors_as_string, true );
-			foreach ( $all_contributors as $key => $val ) {
-				$book_information[ $key ] = $val;
-			};
+			$book_information = array_merge(
+				$book_information,
+				$contributors->getAll(
+					$meta_post->ID,
+					$contributors_as_string,
+					true
+				)
+			);
```

Comment:
It is a simpler way to do what we did previously to store the contributors information in `$book_information`